File: src/main/java/FizzBuzz.java
public class FizzBuzz {
    public String convert(String input) {

        String result = "";

        int inputNumber = Integer.parseInt(input);

        if (inputNumber % 3 == 0) {
            result += "Fizz";
        }

        if (inputNumber % 5 == 0) {
            result += "Buzz";
        }

        if (inputNumber % 7 == 0) {
            result += "Whizz";
        }

        return result.isEmpty() ? input : result;
    }
}

File: src/main/Haskell/FizzBuzz.hs
say a
	| result /= [] = unwords result
	| otherwise = show a
	where result = [output|input <- [a], input `mod` 3 == 0, let output = "Fizz"] ++
				   [output|input <- [a], input `mod` 5 == 0, let output = "Buzz"] ++
				   [output|input <- [a], input `mod` 7 == 0, let output = "Whizz"]